tests/keras/utils/data_utils_test.py

```diff
@@ -245,5 +245,40 @@ def test_ordered_enqueuer_fail_processes():
         next(gen_output)
 
 
+@threadsafe_generator
+def create_finite_generator_from_sequence_threads(ds):
+    for i in range(len(ds)):
+        yield ds[i]
+
+
+def create_finite_generator_from_sequence_pcs(ds):
+    for i in range(len(ds)):
+        yield ds[i]
+
+
+def test_finite_generator_enqueuer_threads():
+    enqueuer = GeneratorEnqueuer(create_finite_generator_from_sequence_threads(
+        TestSequence([3, 200, 200, 3])), use_multiprocessing=False)
+    enqueuer.start(3, 10)
+    gen_output = enqueuer.get()
+    acc = []
+    for output in gen_output:
+        acc.append(int(output[0, 0, 0, 0]))
+    assert len(set(acc) - set(range(100))) == 0, "Output is not the same"
+    enqueuer.stop()
+
+
+def test_finite_generator_enqueuer_processes():
+    enqueuer = GeneratorEnqueuer(create_finite_generator_from_sequence_pcs(
```

Comment:
Please test that the StopIterator is raised

Reply:
@Dref360 Is not this tested directly in the for loop? The StopIterator causes the for loop to exit. Do you mean place a next() call after the end of the loop to show it throws an exception?

Reply:
Oh yeah you're right sorry.